enterprise_access/apps/api_client/lms_client.py

```diff
@@ -100,8 +100,10 @@ def get_enterprise_customer_data(self, enterprise_customer_uuid=None, enterprise
             response = self.client.get(endpoint, timeout=settings.LMS_CLIENT_TIMEOUT)
             response.raise_for_status()
             payload = response.json()
-            if results := payload.get('results'):
-                return results[0]
+            if 'count' in payload:
+                if results := payload.get('results'):
+                    return results[0]
+                return {}
```

Comment:
Do you not also have to check if the count is >= 1?

Reply:
the `if results` part is effectively doing that. If the count is 0, `results` will be an empty list, and thus falsey, so we'd fall into the `return {}` path. The `test_get_enterprise_customer_data_no_hits` test covers this case.